### PR TITLE
 [Feature] Add OpenLayers to Lizmap's map and TreeView

### PR DESCRIPTION
**Pull request by rldhont**
Example:
```js
const extGroupMapState = lizMap.mainLizmap.state.rootMapGroup.createExternalGroup('Test')
const olLayer = new lizMap.ol.layer.Tile({
  source: new lizMap.ol.source.TileWMS({
    url: 'https://ahocevar.com/geoserver/gwc/service/wms',
    crossOrigin: '',
    params: {
      'LAYERS': 'ne:NE1_HR_LC_SR_W_DR',
      'TILED': true,
      'VERSION': '1.1.1',
    },
    projection: 'EPSG:4326',
    // Source tile grid (before reprojection)
    tileGrid: lizMap.ol.tilegrid.createXYZ({
      extent: [-180, -90, 180, 90],
      maxResolution: 360 / 512,
      maxZoom: 10,
    }),
    // Accept a reprojection error of 2 pixels
    reprojectionErrorThreshold: 2,
  }),
});

extGroupMapState.addOlLayer('wms4326', olLayer);

const olLayer2 = new lizMap.ol.layer.Image({
    extent: [-13884991, 2870341, -7455066, 6338219],
    source: new lizMap.ol.source.ImageWMS({
      url: 'https://ahocevar.com/geoserver/wms',
      params: {'LAYERS': 'topp:states'},
      ratio: 1,
      serverType: 'geoserver',
    }),
  });
extGroupMapState.addOlLayer('states', olLayer2);
```

It is possible to remove OpenLayers layer
```js
extGroupMapState.removeOlLayer('wms4326');
```

To clean the external group
```js
extGroupMapState.clean();
```

To remove the external group
```js
lizMap.mainLizmap.state.rootMapGroup.removeExternalGroup('Test');
```

Capture d’écran 2024-04-26 à 18 21 40

Capture d’écran 2024-04-26 à 18 21 45

### CODE REVIEW

File: assets/src/components/Treeview.js [hunk 1]
```diff
@@ -27,9 +26,42 @@ export default class Treeview extends HTMLElement {
...
+                    </div>
+                </div>
+            </div>
+        </li>`
+
+        this._extGroupTemplate = (extGroup) =>
+            html`
+        <li data-testid="ext-${extGroup.name}">
+            <div class="expandable ${extGroup.expanded ? 'expanded' : ''}" @click=${() => extGroup.expanded = !extGroup.expanded}></div>
+            <div class="${extGroup.checked ? 'checked' : ''} ${extGroup.type} group">
```
Comment (nboisteault) [inline assets/src/components/Treeview.js:51]:
```suggestion
            <div class="${extGroup.checked ? 'checked' : ''}  group">
```

**Comment by nboisteault** [comment]
@rldhont it need the legend and it seems good for me.

File: assets/src/components/Treeview.js (hunk 1 above)
Comment (nboisteault) [reply assets/src/components/Treeview.js:51]: It seems `${extGroup.type}` = ext-group. If yes you can ignore my suggestion.

Comment (rldhont) [reply assets/src/components/Treeview.js:51]: extGroup.type is 'ext-group'

**Review by mdouchin — APPROVED** [review]

**Pull request merged**